src/hotspot/share/code/relocInfo.hpp

```diff
@@ -603,29 +603,29 @@ class RelocIterator : public StackObj {
   RelocIterator(nmethod* nm, address begin = nullptr, address limit = nullptr);
   RelocIterator(CodeSection* cb, address begin = nullptr, address limit = nullptr);
   RelocIterator(CodeBlob* cb);
+  RelocIterator(relocInfo& ri);
```

Comment:
How about making this new ctor private?

Reply:
We could do that, as both `Relocation` (the user of the ctr) and `RelocIterator` are friends.